riff-raff/app/deployment/model.scala

```diff
@@ -17,7 +16,12 @@ case object ContinuousDeploymentRequestSource extends RequestSource
 case class ApiRequestSource(key: ApiKey) extends RequestSource
 case object ScheduleRequestSource extends RequestSource
 
-case class Error(message: String) extends AnyVal
+sealed trait ScheduledDeployError
+case class NoDeploysFoundForStage(projectName: String, stage: String) extends ScheduledDeployError
+case class SkippedDueToPreviousFailure(failedDeployRecord: Record) extends ScheduledDeployError
+case class SkippedDueToPreviousWaitingDeploy(waitingDeployRecord: Record) extends ScheduledDeployError
+
+case class Error(message: String, scheduledDeployError: Option[ScheduledDeployError] = None)
```

Comment:
I would rearrange this into an error hierarchy such that Error is a trait with a message and all the other errors extend from it setting the message appropriately. This will make the Error hierarchy more generic so it can be adopted elsewhere as required.

Reply:
> I would rearrange this into an error hierarchy such that Error is a trait with a message and all the other errors extend from it setting the message appropriately.

Thanks for the suggestion, I've tried to do that here: https://github.com/guardian/riff-raff/pull/614/commits/4c70adbf78fee91f6f88e7bea4cc120b6c61d932. 

The `Error(message: String)` case class is already used widely throughout the codebase, so I've given the trait a different name to avoid conflicts!